src/compile/axis/config.ts

```diff
@@ -1,14 +1,71 @@
+import {ScaleType} from 'vega-typings/types';
+import {array} from 'vega-util';
 import {Axis} from '../../axis';
+import {PositionScaleChannel} from '../../channel';
 import {Config} from '../../config';
+import {isQuantitative} from '../../scale';
+import {titlecase} from '../../util';
 import {getStyleConfig} from '../common';
 
+function getAxisConfigFromConfigTypes(configTypes: string[], config: Config) {
+  // TODO: add special casing to add conditional value based on orient signal
+  return Object.assign.apply(null, [{}, ...configTypes.map(configType => config[configType])]);
+}
+
+export type AxisConfigs = ReturnType<typeof getAxisConfigs>;
+
+export function getAxisConfigs(channel: PositionScaleChannel, scaleType: ScaleType, orient: string, config: Config) {
+  const typeBasedConfigTypes =
```

Comment:
convert to if?

Reply:
The chain is really hard to read.

Reply:
nah, I disagree for this one.

Reply:
Why do you think this ternary is better than some ifs?